### PR TITLE
[FabricRuntime] impl get code package info from activation context

### PR DESCRIPTION
**Pull request by youyuanwu**
code package info like work dir, log dir etc can be retrieved from activation context.
activation context is refactored to separate mod.

### CODE REVIEW

File: crates/libs/core/src/types/runtime/mod.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use crate::HSTRING;
+
+#[derive(Debug)]
+pub struct EndpointResourceDesc {
+    pub name: HSTRING,
+    pub protocol: HSTRING,
+    pub r#type: HSTRING,
+    pub port: u32,
+    pub certificate_name: HSTRING,
+    //pub Reserved: *mut ::core::ffi::c_void,
```
Comment (rsajwani) [inline crates/libs/core/src/types/runtime/mod.rs:19]: future use??

**Review by rsajwani — APPROVED** [review]

File: crates/libs/core/src/runtime/activation_context.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+use crate::{strings::HSTRINGWrap, types::EndpointResourceDesc, Error, HSTRING, PCWSTR};
+
+use super::config::ConfigurationPackage;
+
+pub struct CodePackageActivationContext {
+    com_impl: IFabricCodePackageActivationContext6,
+}
+
+/// Info from the context
```
Comment (jrahman) [inline crates/libs/core/src/runtime/activation_context.rs:16]:
```
/// Code package information is propagated here from the ActivationContext
/// provided when the service is started. This information can be used to setup
/// log directories and other resources needed which require information
/// about the code package and the local runtime environment.
```

File: crates/samples/echomain/src/main.rs [hunk 3]
```diff
@@ -39,11 +39,14 @@ fn main() -> mssf_core::Result<()> {
...
         wait_for_debugger();
     }
-    let actctx = ActivationContext::create().inspect_err(|e| {
+    let actctx = CodePackageActivationContext::create().inspect_err(|e| {
         error!("Fail to create activation context: {e}");
     })?;
     validate_configs(&actctx);
 
+    let code_info = actctx.get_code_package_info();
+    info!("code package info: {:?}", code_info);
```
Comment (jrahman) [inline crates/samples/echomain/src/main.rs:48]: Debugging?

**Review by jrahman — APPROVED** [review]

File: crates/libs/core/src/types/runtime/mod.rs (hunk 1 above)
Comment (youyuanwu) [reply crates/libs/core/src/types/runtime/mod.rs]: Removed.

File: crates/samples/echomain/src/main.rs (hunk 3 above)
Comment (youyuanwu) [reply crates/samples/echomain/src/main.rs:48]: Example app output info is ok.

**Pull request merged**